poc_iot_injector/src/server.rs

```diff
@@ -19,29 +18,30 @@ pub const DEFAULT_LAST_POC_SUBMISSION_TS_SECS: i64 = 0000000000;
 
 pub struct Server {
     pool: Pool<Postgres>,
-    keypair: Keypair,
-    txn_service: TransactionService,
+    keypair: Arc<Keypair>,
+    txn_service: Arc<Mutex<TransactionService>>,
     iot_verifier_store: FileStore,
     last_poc_submission_ts: MetaValue<i64>,
 }
 
 impl Server {
     pub async fn new(pool: Pool<Postgres>, keypair: Keypair) -> Result<Self> {
-        // TODO: Find this time in meta store db first
-        let last_poc_submission_ts = env::var("LAST_POC_SUBMISSION_TS")
-            .unwrap_or_else(|_| DEFAULT_LAST_POC_SUBMISSION_TS_SECS.to_string())
-            .parse()
-            .map_err(DecodeError::from)?;
+        // Check meta for last_poc_submission_ts, if not found, use the env var and insert it
+        let last_poc_submission_ts =
+            MetaValue::<i64>::fetch_or_insert_with(&pool, "last_reward_end_time", || {
+                env::var("LAST_POC_SUBMISSION_TS")
+                    .unwrap_or_else(|_| DEFAULT_LAST_POC_SUBMISSION_TS_SECS.to_string())
+                    .parse()
+                    .unwrap_or_default()
```

Comment:
would erroring out here not be more appropriate than defaulting to type default ?

Reply:
+1 - type default is a smell and kind of a dubious concept in general. If anything, use an explicit, domain-specific default value.

Reply:
Updated 00c37db. cc: @madninja there's an expect here now, I think its okay to have it since we do want it to blow up if the `LAST_POC_SUBMISSION` is not found or is set incorrectly.

Reply:
Thanks for not using expect 👍🏻